app/v2/services/XmlParsers.scala

```diff
@@ -20,24 +20,30 @@ import akka.NotUsed
 import akka.stream.alpakka.xml.ParseEvent
 import akka.stream.alpakka.xml.StartElement
 import akka.stream.scaladsl.Flow
+import v2.models.MovementType
 import v2.models.errors.ExtractionError
 import v2.models.request.MessageType
 
 object XmlParsers {
 
-  val messageTypeExtractor: Flow[ParseEvent, Either[ExtractionError, MessageType], NotUsed] = Flow[ParseEvent]
+  def messageTypeExtractor(movementType: MovementType): Flow[ParseEvent, Either[ExtractionError, MessageType], NotUsed] = Flow[ParseEvent]
     .filter {
       case _: StartElement =>
         true
       case _ => false
     }
     .take(1)
     .map {
-      case s: StartElement =>
+      case s: StartElement if movementType == MovementType.Departure =>
```

Comment:
Instead of checking the movementType for every element that comes through, can we set a val here and reduce the duplication. For example:

`val messageTypes = if (movementType == MovementType.Departure) MessageType.messageTypesSentByDepartureTrader else MessageType.messageTypesSentByArrivalTrader`

`case s: StartElement => messageTypes.find(_.rootNode == s.localName)
          .map(Right(_))
          .getOrElse(Left(ExtractionError.MessageTypeNotFound(s.localName))) `

Reply:
@drnaylor-oc got there before me!

Reply:
done as suggested